components/admin_console/team_channel_settings/channel/list/channel_list.tsx

```diff
@@ -2,182 +2,227 @@
-                            defaultMessage='Management'
+                            id={`admin.channel_settings.channel_row.managementMethod.${channel.group_constrained ? 'group' : 'manual'}`}
+                            defaultMessage={channel.group_constrained ? 'Group Sync' : 'Manual Invites'}
                         />
-                    </div>
-                    <div className='group-actions'/>
-                </div>
-            </div>
-        );
+                    </span>
+                ),
+                edit: (
+                    <span
+                        className='group-actions TeamList_editRow'
+                        data-testid={`${channel.display_name}edit`}
+                    >
+                        <Link to={`/admin_console/user_management/channels/${channel.id}`} >
+                            <FormattedMessage
+                                id='admin.channel_settings.channel_row.configure'
+                                defaultMessage='Edit'
+                            />
+                        </Link>
+                    </span>
+                ),
+            };
+        });
     }
 
-    onPageChangedCallback = (pagination: {startCount: number; endCount: number; total: number}, channels: ChannelWithTeamData[]) => {
-        if (this.state.searchMode) {
-            this.setState({channels});
-        }
-    };
+    render = (): JSX.Element => {
+        const {term} = this.state;
+        const rows: Row[] = this.getRows();
+        const columns: Column[] = this.getColumns();
+        const {startCount, endCount, total} = this.getPaginationProps();
 
-    render(): JSX.Element {
-        const absProps = {...this.props};
-        if (this.state.searchMode) {
-            absProps.actions.getData = this.getDataBySearch;
-        }
-        return (
-            <div className='groups-list groups-list-no-padding'>
-                {this.searchBar()}
-                <AbstractList
-                    header={this.header()}
-                    renderRow={this.renderRow}
-                    noPadding={true}
-                    {...absProps}
-                    key={this.state.pageResetKey}
-                    onPageChangedCallback={this.onPageChangedCallback}
-                    data={this.state.searchMode ? this.state.channels : this.props.data}
-                    total={this.state.searchMode ? this.state.searchTotalCount : this.props.total}
-                />
-            </div>
+        const placeholderEmpty: JSX.Element = (
+            <FormattedMessage
+                id='admin.channel_settings.channel_list.no_channels_found'
+                defaultMessage='No channels found'
+            />
         );
-    }
 
-    private renderRow = (item: ChannelWithTeamData) => {
         return (
-            <ChannelRow
-                key={item.id}
-                channel={item}
-                onRowClick={this.onChannelClick}
```

Comment:
in order to prevent jittery behaviour when changing pages you'll need to do something similar to what I did in `user_grid` line 267
```
        const rowsContainerStyles = {
            minHeight: `${rows.length * ROW_HEIGHT}px`,
        };
```

And then pass `rowsContainerStyles` to `DataGrid` this prevents the datagrid from shrinking down while its loading between pages i.e. it retains its previous grid height when its loading the next page

Reply:
![Kapture 2020-06-11 at 20 49 22](https://user-images.githubusercontent.com/3207297/84452898-138d0d80-ac25-11ea-82dc-f8e73f9795bb.gif)
Heres a gif showing what I mean